From c4874fb8866046ea62731399bc58740caeb154b7 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Wed, 12 Feb 2025 13:23:27 +0100
Subject: [PATCH] fix MDI-X showing as Unknown instead of "off (auto)"

The current version of ethtool is unable to show the correct MDI-X info:
 # ethtool --version
 ethtool version 6.11
 # ethtool enp0s31f6 | grep MDI
     MDI-X: Unknown

For comparison, an older version shows it correctly:
 # ./ethtool --version
 ethtool version 6.2
 # ./ethtool enp0s31f6 | grep MDI
     MDI-X: off (auto)

The blamed commit accidentally removed the ETH_TP_MDI switch case
in dump_mdix(). As a result, ETH_TP_MDI is treated like Unknown.

Fix it by restoring the ETH_TP_MDI case and breaking out.
'mdi_x' is initialized to false, which is correct for this case.

Fixes: bd1341cd2146 ("add json support for base command")
Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
---
 common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common.c b/common.c
index 4fda4b49d2fd..1ba27e7577b4 100644
--- a/common.c
+++ b/common.c
@@ -171,6 +171,8 @@ void dump_mdix(u8 mdix, u8 mdix_ctrl)
 		mdi_x_forced = true;
 	} else {
 		switch (mdix) {
+		case ETH_TP_MDI:
+			break;
 		case ETH_TP_MDI_X:
 			mdi_x = true;
 			break;
-- 
2.49.0

